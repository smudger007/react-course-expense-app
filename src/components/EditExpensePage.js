import React from 'react';
import {connect} from 'react-redux'
import ExpenseForm from './ExpenseForm';

const EditExpensePage = (props) => {
    console.log(props);
    return (
        <ExpenseForm
            expense={props.expense}
            onSubmit={(expense) => {                       
                console.log('expense');
            }}    
        />
    );
};

const mapStateToProps = (state, props) => {
    return {
        expense: state.expenses.find((expense) => expense.id === props.match.params.id)
    };
};

export default connect(mapStateToProps)(EditExpensePage);

