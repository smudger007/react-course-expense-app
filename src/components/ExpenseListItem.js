import React from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import {removeExpense} from '../actions/expenses';
import moment from 'moment';
import numeral from 'numeral';


const ExpenseListItem = ({dispatch, id, description, amount, createdAt}) => (
    <div className='expense'>
        <Link 
            to={`/edit/${id}`}
        >
            <h3>{description}</h3>
        </Link>
        <p>
         {numeral(amount / 100).format('Â£0,0.00')}
         - 
         {moment(createdAt).format('MMMM Do, YYYY')} 
         </p>
        <button
            className="button button--link"
            onClick={(e) => {
                dispatch(removeExpense({id}));
            }}
        >
        Remove
        </button> 
    </div>
);

export default connect()(ExpenseListItem);