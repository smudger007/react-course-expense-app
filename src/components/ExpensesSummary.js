import React from 'react';
import numeral from 'numeral';
import { connect } from 'react-redux';
import selectExpensesTotal from '../selectors/expenses-total'
import selectExpenses from '../selectors/expenses'


export const ExpensesSummary = ({expensesCount, expensesTotal}) => {

    const expensesTerm = expensesCount === 1 ? "expense" : "expenses";
    const formattedAmount = numeral(expensesTotal / 100).format('$0,0.00');

    return (
        <div>
            <h1>Displaying {expensesCount} {expensesTerm} totalling {formattedAmount} </h1>
        </div>
    );
};

const mapStateToProps = (state) => {
    const visibleExpenses = selectExpenses(state.expenses, state.filters);

    return {
        expensesCount: visibleExpenses.length, 
        expensesTotal: selectExpensesTotal(visibleExpenses)
    };
};

export default connect(mapStateToProps)(ExpensesSummary);