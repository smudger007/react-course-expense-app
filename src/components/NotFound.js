import React from 'react';

const NotFound = () => (
    <div>404 Sucker!!!! <Link to="/">Go Home you Muppet!!</Link></div>
);

export default NotFound;