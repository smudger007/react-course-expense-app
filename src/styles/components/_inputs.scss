.text-inputs {
    border: 1px solid #cacccd;
    height: 50px;
    font-size: $font-size-large;
    font-weight: 300;
    padding: $s-size;
}

.select {
    @extend .text-inputs;
}

.text-area {
    @extend .text-inputs;
    height: 10rem;
}