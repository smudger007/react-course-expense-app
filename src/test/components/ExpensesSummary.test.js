import React from 'react';
import { shallow } from 'enzyme';

import { ExpensesSummary } from '../../components/ExpensesSummary';

test('should render ExpensesSummary correctly for 1 expense', () => {

    const wrapper = shallow(<ExpensesSummary expensesCount={1} expensesTotal={234} />);    
    expect(wrapper).toMatchSnapshot();
   
});

test('should render ExpensesSummary correctly for multiple expenses', () => {

    const wrapper = shallow(<ExpensesSummary expensesCount={4} expensesTotal={4354} />);    
    expect(wrapper).toMatchSnapshot();
   
});
