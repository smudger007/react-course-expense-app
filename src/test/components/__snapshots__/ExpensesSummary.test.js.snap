// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpensesSummary correctly for 1 expense 1`] = `
<div>
  <h1>
    Displaying 
    1
     
    expense
     totalling 
    $2.34
     
  </h1>
</div>
`;

exports[`should render ExpensesSummary correctly for multiple expenses 1`] = `
<div>
  <h1>
    Displaying 
    4
     
    expenses
     totalling 
    $43.54
     
  </h1>
</div>
`;
